app/src/main/kotlin/com/wire/android/ui/common/topappbar/search/SearchBarState.kt

```diff
@@ -57,14 +58,23 @@ class SearchBarState(
     }
 
     companion object {
-        fun saver(searchQueryTextState: TextFieldState): Saver<SearchBarState, *> = Saver(
+        fun saver(): Saver<SearchBarState, *> = Saver(
             save = {
-                listOf(it.isSearchActive)
+                listOf(
+                    it.isSearchActive,
+                    with(TextFieldState.Saver) {
+                        save(it.searchQueryTextState)
+                    }
+                )
             },
             restore = {
                 SearchBarState(
-                    isSearchActive = it[0],
-                    searchQueryTextState = searchQueryTextState
+                    isSearchActive = it[0] as Boolean,
+                    searchQueryTextState = it[1]?.let {
```

Comment:
Question: Are there chances of getting an `IndexOutOfBounds` here ? Asking just in case we can make it safer (for example using a Pair instead of a List?), otherwise it looks good to me.

Reply:
In theory, if that is saved properly then it shouldn't make any problems when restoring and even Google does that without any checks but it's a nice suggestion to add that just to be sure in any case 😄 So I changed `it[1]` with `it.getOrNull(1)`